### PR TITLE
Runs integration tests on travis

### PR DESCRIPTION
**Pull request by vijetmahabaleshwar-okta**
* Adds support to run integration tests for PRs
* Will only run ITs for PRs and cron jobs, not for push requests
* If we run ITs for push requests as well, since both builds (push and pull) will be running simultaneously against the same trex org, tests will fail (with resource already exists error)

### CODE REVIEW

File: src/Okta.Sdk/Internal/OktaClientConfigurationValidator.cs [hunk 1]
```diff
@@ -29,7 +29,7 @@ public static void Validate(OktaClientConfiguration configuration)
 
             if (!configuration.OktaDomain.StartsWith("https://", StringComparison.OrdinalIgnoreCase))
             {
-                throw new ArgumentNullException(nameof(configuration.OktaDomain), "Your Okta URL must start with https. You can copy your domain from the Okta Developer Console.");
+                throw new ArgumentException(nameof(configuration.OktaDomain), "Your Okta URL must start with https. You can copy your domain from the Okta Developer Console.");
```
Comment (vijetmahabaleshwar-okta) [inline src/Okta.Sdk/Internal/OktaClientConfigurationValidator.cs]: This change is needed to get the following IT to pass - `Okta.Sdk.IntegrationTests/OktaClientShould.cs`
It expects the `ArgumentException`.

File: src/Okta.Sdk.IntegrationTests/OktaClientShould.cs [hunk 2]
```diff
@@ -71,6 +71,7 @@ public async Task ThrowApiExceptionForInvalidToken()
             var client = TestClient.Create(new OktaClientConfiguration
             {
                 Token = "abcd1234",
```
Comment (laura-rodriguez) [inline src/Okta.Sdk.IntegrationTests/OktaClientShould.cs]: In this test, we shouldn't enable the flag. Instead, we have to add a valid oktaDomain.

File: src/Okta.Sdk.IntegrationTests/LogsScenarios.cs [hunk 3]
```diff
@@ -99,6 +99,13 @@ public async Task GetLogsByEventType()
         [Fact]
         public async Task GetLogsBySinceDate()
         {
+            // Don't run test on travis - https://github.com/okta/okta-sdk-dotnet/issues/271
```
Comment (vijetmahabaleshwar-okta) [inline src/Okta.Sdk.IntegrationTests/LogsScenarios.cs]: This test always fails on travis. I've created a github issue to track it.

**Comment by nbarbettini** [comment]
@vijetmahabaleshwar-okta Let me know when you want me to re-review this. (And I agree with Laura, this branch should be updated/rebased to latest master)

**Comment by laura-rodriguez** [comment]
We would be able to use `NewGuid()` in resources name to avoid:
_If we run ITs for push requests as well, since both builds (push and pull) will be running simultaneously against the same trex org, tests will fail (with resource already exists error)_
But is it the solution @nbarbettini? Other than that, it looks good to me 👍.

File: build.sh [hunk 4]
```diff
@@ -22,9 +22,16 @@ else
     MD5_EXE="md5sum"
 fi
 
-# Define default arguments.
+# Define default arguments
 SCRIPT="build.cake"
-TARGET="Default"
+TARGET=
+# Don't run ITs for push requests
+if [[ $TRAVIS_EVENT_TYPE != 'push' ]]; then
```
Comment (nbarbettini) [inline build.sh]: Why run ITs by default at all? You should be able to manually do:

```
./build.sh --target DefaultIT
```

File: src/Okta.Sdk.IntegrationTests/LogsScenarios.cs [hunk 5]
```diff
@@ -103,6 +103,13 @@ public async Task GetLogsByEventType()
         [Trait("Category", "NoBacon")] // Tests that don't run on internal CI pipeline
         public async Task GetLogsBySinceDate()
         {
+            // Don't run test on travis - https://github.com/okta/okta-sdk-dotnet/issues/271
+            string travis = System.Environment.GetEnvironmentVariable("TRAVIS");
```
Comment (nbarbettini) [inline src/Okta.Sdk.IntegrationTests/LogsScenarios.cs]: Let's `[Skip]` tests that are currently broken rather than hard code env var checks.

Comment (vijetmahabaleshwar-okta) [reply src/Okta.Sdk.IntegrationTests/LogsScenarios.cs]: Done

**Comment by nbarbettini** [comment]
@laura-rodriguez 
> We would be able to use NewGuid() in resources name to avoid:
> _If we run ITs for push requests as well, since both builds (push and pull) will be running simultaneously against the same trex org, tests will fail (with resource already exists error)_
I agree, I should have built them that way (with random GUIDs) from the beginning. Otherwise the IT suite is always a "singleton" with respect to the org. We should be able to run tests in parallel as much as we want (because the tests clean up properly _and_ use randomized emails). Separate PR?

**Pull request merged**